pkg/query-service/app/logparsingpipeline/preview.go

```diff
@@ -0,0 +1,254 @@
+	processorConfigs, err := collectorProcessorsForPipelines(pipelines)
+	if err != nil {
+		return nil, model.BadRequest(errors.Wrap(
+			err, "could not prepare otel processors for pipelines",
+		))
+	}
+
+	processorFactories, err := processor.MakeFactoryMap(
+		logstransformprocessor.NewFactory(),
+	)
+	if err != nil {
+		return nil, model.InternalError(errors.Wrap(
+			err, "could not construct processor factory map",
+		))
+	}
+
+	// Pipelines translate to logtransformprocessors in otel collector config.
+	// Each logtransformprocessor (stanza) does its own batching with a flush
+	// interval of 100ms. So e2e processing time for logs grows linearly with
+	// the number of logtransformprocessors involved.
+	// See defaultFlushInterval at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/adapter/emitter.go
+	// TODO(Raj): Remove this after flushInterval is exposed in logtransformprocessor config
+	timeout := time.Millisecond * time.Duration(len(processorConfigs)*100+100)
+
+	outputPLogs, collectorErrs, apiErr := collectorsimulator.SimulateLogsProcessing(
+		ctx,
+		processorFactories,
+		processorConfigs,
+		simulatorInputPLogs,
+		timeout,
+	)
+	collectorErrsText := strings.Join(collectorErrs, "\n")
+	if apiErr != nil {
+		return nil, model.WrapApiError(apiErr, fmt.Sprintf(
+			"could not simulate log pipelines processing.\nCollector errors: %s\n", collectorErrsText,
+		))
+	}
+
+	outputSignozLogs := PLogsToSignozLogs(outputPLogs)
+
+	// Sort output logs by their order in the input and remove the temp ordering attribute
+	sort.Slice(outputSignozLogs, func(i, j int) bool {
+		iIdx := outputSignozLogs[i].Attributes_int64[inputOrderAttribute]
+		jIdx := outputSignozLogs[j].Attributes_int64[inputOrderAttribute]
+		return iIdx < jIdx
+	})
+	for _, sigLog := range outputSignozLogs {
+		delete(sigLog.Attributes_int64, inputOrderAttribute)
+	}
+
+	return outputSignozLogs, nil
+}
+
+func collectorProcessorsForPipelines(pipelines []Pipeline) (
+	[]collectorsimulator.ProcessorConfig, error,
+) {
+	processors, procNames, err := PreparePipelineProcessor(pipelines)
+	if err != nil {
+		return nil, err
+	}
+
+	processorConfigs := []collectorsimulator.ProcessorConfig{}
+	for _, procName := range procNames {
+		// convert `Processor` structs to map[string]interface{}
+		procYaml, err := yaml.Marshal(processors[procName])
+		if err != nil {
+			return nil, errors.Wrap(err, "could not marshal Processor struct")
+		}
+		var procConfRaw map[string]interface{}
+		err = yaml.Unmarshal(procYaml, &procConfRaw)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not unmarshal proc yaml")
+		}
+
+		processorConfigs = append(processorConfigs, collectorsimulator.ProcessorConfig{
+			Name:   procName,
+			Config: procConfRaw,
+		})
+	}
+
+	return processorConfigs, nil
+}
+
+// plog doesn't contain an ID field.
+// SignozLog.ID is stored as a log attribute in plogs for processing
+// and gets hydrated back later.
```

Comment:
The `ID` is generated by us dynamically for each record. It looks like this will be missing? Does the frontend work without ID?

Reply:
I think this is fine. I had a different assumption. I thought the user would simply paste the logs from their application instead of UI using the logs from the query range.